hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -399,15 +400,48 @@ private void updateKeys(
             validateTrue(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
         }
         if (op.hasAdminKey()) {
+            removeKeysIfNeeded(op, builder);
             final var newAdminKey = op.adminKey();
-            if (isKeyRemoval(newAdminKey)) {
-                builder.adminKey((Key) null);
-            } else {
+            if (!isKeyRemoval(newAdminKey)) {
                 builder.adminKey(newAdminKey);
             }
         }
     }
 
+    private void removeKeysIfNeeded(final TokenUpdateTransactionBody op, final Token.Builder builder) {
+        if (IMMUTABILITY_SENTINEL_KEY.equals(op.adminKey())) {
```

Comment:
all occurences of `IMMUTABILITY_SENTINEL_KEY.equals(*some_key*)` in `removeKeysIfNeeded` can be replaced with `isKeyRemoval(*some_key*)`

Reply:
Added, thanks